analyzer/item/item.go

```diff
@@ -154,7 +154,14 @@ func (a *itemBasedAnalyzer[Item]) processBatch(ctx context.Context) (int, error)
 	// Commit the changes from all successfully processed items to the database.
 	batch := &storage.QueryBatch{}
 	for _, b := range batches {
+		// Each item's batch is wrapped in a transaction because:
+		//  - This makes txs smaller, reducing the impact of deadlocks.
+		//  - We can specify a high isolation level. Item analyzers run in parallel with block analyzers,
+		//    and might need a consistent view of the database. E.g. the `evm_token_balances` analyzer
+		//    needs to read the latest height and the balance at that height.
+		batch.Queue("BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE")
 		batch.Extend(b)
+		batch.Queue("COMMIT")
 	}
```

Comment:
This change is somewhat unrelated, and will have a much higher impact on our potential for deadlocks.
I think we don't need this for the general item analyzers, and not even for the evm balances analyzer.
But I wanted to show the option here, and hear comments. For now, I plan to remove this from the PR before merging.

If we do merge this, we'll have to evaluate carefully, as it might tank performance.

Reply:
this here as an anchor for discussion. we can't literally do this because our send-batch-fast-retry-on-error is hardcoded for complete rollback on error

Reply:
I believe it would work if send-batch-fast used explicit txs, rather than the pipeline stuff. With an explicit tx, this change would produce nested txs, whereas send-batch-fast would roll back the top-level tx.
OTOH the large parent tx would also negate the proposed benefits here.

I'm getting annoyed with how the pipeline mode needs special consideration. I'll remove (= always start an explicit tx) it in a separate PR.